service/iot-service-samples/pnp-service-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/TemperatureController.java

```diff
@@ -0,0 +1,143 @@
+package samples.com.microsoft.azure.sdk.iot;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.microsoft.azure.sdk.iot.service.devicetwin.*;
+import com.microsoft.azure.sdk.iot.service.exceptions.IotHubException;
+
+import javax.json.Json;
+import java.io.IOException;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+// This sample uses the model - https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/TemperatureController.json
+public class TemperatureController {
+    // Get connection string and device id inputs
+    private static final String hubConnectionString  = "HostName=summerrelease-test-04.private.azure-devices-int.net;SharedAccessKeyName=iothubowner;SharedAccessKey=lkm4Gvm5gN9nWDgK4Cwwwgj47q/8w1qxgF5bRbi6yNM=";
+    private static final String deviceId = "vinageshpnpdevice1";
+
+    private static DeviceTwin twinClient;
+    private static DeviceMethod methodClient;
+
+    public static void main(String[] args) throws Exception {
+        RunSample();
+    }
+
+    private static void RunSample() throws IOException, IotHubException {
+        System.out.println("Initialize the service client.");
+        InitializeServiceClient();
+
+        System.out.println("Get Twin model Id and Update Twin");
+        GetAndUpdateTwin();
+
+        System.out.println("Invoke a method on component");
+        InvokeMethodOnComponent();
+
+        System.out.println("Invoke a method on root level");
+        InvokeMethodOnRootLevel();
+    }
+
+    private static void InitializeServiceClient() throws IOException {
+        twinClient = DeviceTwin.createFromConnectionString(hubConnectionString);
+        methodClient = DeviceMethod.createFromConnectionString(hubConnectionString);
+    }
+
+    private static void GetAndUpdateTwin() throws IOException, IotHubException {
+        // Get a Twin and retrieves model Id set by Device client
+        DeviceTwinDevice twin = new DeviceTwinDevice(deviceId);
+        twinClient.getTwin(twin);
+        System.out.println("Model Id of this Twin is: " + twin.getModelId());
+
+        // Update the twin - Update the thermostat1 component
+        twin.setDesiredProperties(CreatePropertyPatch());
+        System.out.println("Updating Device twin property");
+        twinClient.updateTwin(twin);
+
+        twinClient.getTwin(twin);
+        System.out.println("The updated desired properties: " + twin.getDesiredProperties().iterator().next().getValue());
+    }
+
+    private static Set<Pair> CreatePropertyPatch()
```

Comment:
I was not very clear if helper meant helper class or method. Will move it to a class if that is what it means.

Reply:
Based on discussion, these need to be in a helper class so I will move them.